core/src/main/java/org/axonframework/commandhandling/model/inspection/AggregateMemberAnnotatedChildEntityMapDefinition.java

```diff
@@ -17,73 +16,55 @@
-                                   "Please provide type explicitly in @AggregateMember annotation.",
-                           field.toGenericString())));
-        }
+public class AggregateMemberAnnotatedChildEntityMapDefinition extends AbstractChildEntityCollectionDefinition {
 
-        return entityType;
+    @Override
+    protected Optional<Class<?>> resolveGenericType(Field field) {
+        return ReflectionUtils.resolveGenericType(field, 1);
     }
 
     @Override
-    @SuppressWarnings("unchecked")
     public <T> Optional<ChildEntity<T>> createChildDefinition(Field field, EntityModel<T> declaringEntity) {
-        Map<String, Object> attributes =
-                AnnotationUtils.findAnnotationAttributes(field, AggregateMember.class).orElse(null);
+        Map<String, Object> attributes = findAnnotationAttributes(field, AggregateMember.class).orElse(null);
         if (attributes == null || !Map.class.isAssignableFrom(field.getType())) {
             return Optional.empty();
         }
-
         EntityModel<Object> childEntityModel = declaringEntity.modelOf(resolveType(attributes, field));
-        Map<String, Property<Object>> routingKeyProperties = childEntityModel.commandHandlers().values().stream()
-                .map(h -> h.unwrap(CommandMessageHandlingMember.class).orElse(null)).filter(Objects::nonNull)
-                .collect(Collectors.toMap(CommandMessageHandlingMember::commandName, h -> {
-                    String routingKey = getOrDefault(h.routingKey(), childEntityModel.routingKey());
-                    Property<Object> property = getProperty(h.payloadType(), routingKey);
-                    if (property == null) {
-                        throw new AxonConfigurationException(
-                                format("Command of type [%s] doesn't have a property matching the routing key [%s] " +
-                                               "necessary to route through field [%s]",
-                                       h.payloadType(), routingKey, field.toGenericString()));
-                    }
-                    return property;
-                }));
-        return Optional.of(new AnnotatedChildEntity<>(childEntityModel, (Boolean) attributes.get("forwardCommands"),
-                                                      (Boolean) attributes.get("forwardEvents"), (msg, parent) -> {
-            Object routingValue = routingKeyProperties.get(msg.getCommandName()).getValue(msg.getPayload());
-            Map<?, ?> fieldValue = ReflectionUtils.getFieldValue(field, parent);
-            return fieldValue == null ? null : fieldValue.get(routingValue);
-        }, (msg, parent) -> {
-            Map<?, Object> fieldValue = ReflectionUtils.getFieldValue(field, parent);
-            return fieldValue == null ? Collections.emptyList() : fieldValue.values();
-        }));
 
+        Map<String, Property<Object>> routingKeyProperties = extractCommandHandlerRoutingKeys(field, childEntityModel);
```

Comment:
I'd suggest to rename to ``commandRoutingKeyProperties``

Reply:
Agreed, changed.